components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/ClientCertificateBasedAuthenticationHandler.java

```diff
@@ -127,4 +166,28 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext)
         return authenticationResult;
     }
 
+    private boolean requireIntermediateCertValidation(AuthenticationContext authenticationContext) {
+
+        if (!AuthConfigurationUtil.getInstance().isIntermediateCertValidationEnabled()) {
+            return false;
+        }
+        for (String context : AuthConfigurationUtil.getInstance().getExemptedContextList()) {
+            if (authenticationContext.getAuthenticationRequest().getContextPath().contains(context)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private String getCN(String dn) throws InvalidNameException {
```

Comment:
Add method level comment

Reply:
Added in https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/77